src/qibo/core/cgates.py

```diff
@@ -50,16 +48,18 @@ def reprepare(self):
     def prepare(self):
         """Prepares the gate for application to state vectors."""
         self.is_prepared = True
```

Comment:
Do we really need prepare at this point?

Reply:
I think no. I would like to remove `prepare` completely, I am just planning to do it after "merging" `gates` and `cgates` so that it works for all backends.

Reply:
Ok, great.